### PR TITLE
feat: Automate CMP Testing

### PR DESCRIPTION
**Pull request by akinsola-guardian**
## What does this change?
- Automates CMP tests using AWS Eventbridge for scheduled triggers.
- Refactored allow user to trigger lambda in different environments and jurisidictions.

## Why?
When we deploy the CMP to CODE we want to test if it is behaving as expected, we currently do this manually. However there are automated tests against PROD in the form of the "commercial canaries".

We should adapt the current tests to run against the CODE environment on demand to improve speed and reliability of CMP deployment.

### CODE REVIEW

File: monitoring/src/config.ts [hunk 1]
```diff
@@ -106,19 +84,53 @@ const availableEnvConfig = [
...
+	async run(): Promise<void> {
+		this.generateConfig();
+		if (this._config) {
+			await this._config.checkFunction(this._config);
+		}
 	}
 
-	return foundConfig;
-})();
+	private generateConfig(): void {
```
Comment (kenoir) [inline monitoring/src/config.ts:111]: Very nice. Maybe make it log what it's doing i.e. `console.log("Running in Scheduled mode")` or something like that.

File: monitoring/src/utils/config-helper.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	AP_SOUTHEAST_2: 'ap-southeast-2',
+};
+
+export const JURISDICTIONS = {
+	TCFV2: 'tcfv2',
+	CCPA: 'ccpa',
+	AUS: 'aus',
+};
+
+export class ConfigHelper {
```
Comment (kenoir) [inline monitoring/src/utils/config-helper.ts:16]: Does this need to be a class or can we export the functions without the class wrapper?

File: monitoring/src/utils/logger.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
+class Logger {
```
[the remark this answers is not in the document]
Comment (kenoir) [reply monitoring/src/utils/logger.ts:1]: There is also this in [`guardian/libs`](https://github.com/guardian/libs/blob/main/src/logger/README.md) but that is intended as a client side logger (linked here for reference as it's still interesting).

**Comment by akinsola-guardian** [comment]
[Run the lambdas on a schedule](https://trello.com/c/wcRBLzRl/708-run-the-lambdas-on-a-schedule)

**Pull request closed without merging**